src/main/java/com/gradle/develocity/bamboo/SystemProperty.java

```diff
@@ -1,11 +1,69 @@
 package com.gradle.develocity.bamboo;
 
-final class SystemProperty {
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collection;
+
+public final class SystemProperty {
+
+    enum SimpleSystemPropertyKey {
+        MAVEN_EXT_CLASS_PATH_SYSTEM_PROPERTY("maven.ext.class.path");
+
+        SimpleSystemPropertyKey(String key) {
+            this.key = key;
+        }
+
+        public SystemProperty forValue(String value) {
+            return new SystemProperty(key, value);
+        }
+
+        private final String key;
+    }
+
+    enum SystemPropertyKeyWithDeprecatedKey {
+
+        UPLOAD_IN_BACKGROUND_SYSTEM_PROPERTIES("develocity.scan.uploadInBackground", "gradle.scan.uploadInBackground"),
+        SERVER_URL_SYSTEM_PROPERTIES("develocity.url", "gradle.enterprise.url"),
+        ALLOW_UNTRUSTED_SERVER_SYSTEM_PROPERTIES("develocity.allowUntrustedServer", "gradle.enterprise.allowUntrustedServer");
+
+        private final String develocityKey;
+
+        @Nullable
+        private final String deprecatedKey;
+
+        SystemPropertyKeyWithDeprecatedKey(String develocityKey, @Nullable String deprecatedKey) {
+            this.develocityKey = develocityKey;
+            this.deprecatedKey = deprecatedKey;
+        }
+
+        public Collection<SystemProperty> forValue(String value) {
+            return propertiesFor(value);
+        }
+
+        public Collection<SystemProperty> forValue(boolean value) {
+            return propertiesFor(Boolean.toString(value));
+        }
+
+        private Collection<SystemProperty> propertiesFor(@Nullable String value) {
+            Collection<SystemProperty> properties = new ArrayList<>();
+
+            String val = (value == null) ? "" : value;
+
+            properties.add(new SystemProperty(develocityKey, val));
+            if (deprecatedKey != null) {
```

Comment:
If you already support omitting the deprecated key, why do you still have `SimpleSystemPropertyKey`?

Reply:
Good point. I guess I changed my mind during the implementation. :)
I will make this non-nullable here.

Reply:
DONE (168b835d)